src/config.rs

```diff
@@ -82,6 +82,50 @@ macro_rules! create_config {
             $(pub $i: Option<$ty>),+
         }
 
+        // This trait and the following impl blocks are there only so that we
+        // can use UCFS inside the get_docs() function on builtin types for configs.
+        trait IsConfigType {
+            fn get_variant_names() -> Vec<&'static str>;
```

Comment:
Would it be okay for it to return a `String` instead?  I otherwise run into issues in the impl for `get_variant_names()` for the enums in `impl_enum_decodable!` because I was creating a `Vec<&'static str'>`; I can't then do `join().as_str()` because the resulting `&'static str` wouldn't live long enough